kv/kvserver: TestBackpressureNotAppliedWhenReducingRangeSize failed

[(kv/kvserver).TestBackpressureNotAppliedWhenReducingRangeSize failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2744481&tab=buildLog) on [master@ad5662b2fce61455609f1a50bc73c7b4b6e63276](https://github.com/cockroachdb/cockroach/commits/ad5662b2fce61455609f1a50bc73c7b4b6e63276):

```
=== RUN   TestBackpressureNotAppliedWhenReducingRangeSize
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestBackpressureNotAppliedWhenReducingRangeSize721584178
    test_log_scope.go:74: use -show-logs to present logs inline
=== CONT  TestBackpressureNotAppliedWhenReducingRangeSize
    client_replica_backpressure_test.go:304: -- test log scope end --
E210304 10:14:59.393752 486147 kv/kvserver/queue.go:1093  [n2,split,s2,r37/2:/{Table/52-Max}] 7  split at key /Table/52/1/2 failed: fetching current range descriptor value: failed to send RPC: sending to all replicas failed; last error: node unavailable; try another peer
    client_replica_backpressure_test.go:304: Leaked goroutine: goroutine 485665 [chan receive]:
        github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop.func1(0x54d1960, 0xc00bb10d80)
        	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/table_cache.go:176 +0x76
        runtime/pprof.Do(0x54d1960, 0xc00bb10d80, 0xc00069c800, 0x1, 0x1, 0xc001b17fb8)
        	/usr/local/go/src/runtime/pprof/runtime.go:40 +0xcc
        github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop(0xc0072ebd00)
        	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/table_cache.go:175 +0x89
        created by github.com/cockroachdb/pebble.(*tableCacheShard).init
        	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/table_cache.go:167 +0x1a5
--- FAIL: TestBackpressureNotAppliedWhenReducingRangeSize (8.56s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestBackpressureNotAppliedWhenReducingRangeSize PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestBackpressureNotAppliedWhenReducingRangeSize.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

Nevermind here we go:

```
=== CONT  TestBackpressureNotAppliedWhenReducingRangeSize
    client_replica_backpressure_test.go:304: -- test log scope end --
panic: pebble: closed

goroutine 6607 [running]:
github.com/cockroachdb/pebble.(*DB).newIterInternal(0xc00207b800, 0x0, 0x0, 0xc0039a83c8, 0xc00287ba00)
	/home/tobias/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/db.go:690 +0x345
github.com/cockroachdb/pebble.(*DB).NewIter(0xc00207b800, 0xc0039a83c8, 0x8)
	/home/tobias/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/db.go:848 +0x3d
github.com/cockroachdb/cockroach/pkg/storage.(*pebbleIterator).init(0xc0039a83c0, 0x5465d60, 0xc00207b800, 0x0, 0x0, 0x0, 0xc00287b9e0, 0x7, 0x7, 0xc00287b9e8, ...)
	/home/tobias/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble_iterator.go:170 +0x4a4
github.com/cockroachdb/cockroach/pkg/storage.newPebbleIterator(0x5465d60, 0xc00207b800, 0x0, 0x0, 0x0, 0xc00287b9e0, 0x7, 0x7, 0xc00287b9e8, 0x8, ...)
	/home/tobias/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble_iterator.go:90 +0xd8
github.com/cockroachdb/cockroach/pkg/storage.(*Pebble).NewEngineIterator(0xc004f140d0, 0x0, 0xc00287b9e0, 0x7, 0x7, 0xc00287b9e8, 0x8, 0x8, 0x0, 0x0, ...)
	/home/tobias/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble.go:743 +0x78
github.com/cockroachdb/cockroach/pkg/storage.newIntentInterleavingIterator(0x54fc800, 0xc004f140d0, 0x0, 0x73071c7, 0x1, 0x1, 0xc004f44510, 0x2, 0x8, 0x0, ...)
	/home/tobias/go/src/github.com/cockroachdb/cockroach/pkg/storage/intent_interleaving_iter.go:212 +0x235
github.com/cockroachdb/cockroach/pkg/storage.(*intentInterleavingReader).NewMVCCIterator(0xc002878270, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc004f44510, 0x2, 0x8, 0x0, ...)
	/home/tobias/go/src/github.com/cockroachdb/cockroach/pkg/storage/intent_reader_writer.go:299 +0x145
github.com/cockroachdb/cockroach/pkg/storage.(*Pebble).NewMVCCIterator(0xc004f140d0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc004f44510, 0x2, 0x8, 0x0, ...)
	/home/tobias/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble.go:724 +0xe5
github.com/cockroachdb/cockroach/pkg/storage.MVCCFindSplitKey(0x5492920, 0xc002c13e60, 0x54fc800, 0xc004f140d0, 0xc004f44508, 0x1, 0x8, 0xc004f44510, 0x2, 0x8, ...)
	/home/tobias/go/src/github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:3446 +0x196
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).adminSplitWithDescriptor(0xc0009c3b00, 0x5492920, 0xc002c13e60, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/tobias/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_command.go:321 +0x1f0
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*splitQueue).processAttempt(0xc002b2cce0, 0x5492920, 0xc002c13e60, 0xc0009c3b00, 0xc0002ec730, 0x1, 0x2, 0x1)
	/home/tobias/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/split_queue.go:208 +0xb05
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*splitQueue).process(0xc002b2cce0, 0x5492920, 0xc002c13e60, 0xc0009c3b00, 0xc0002ec730, 0x473b348, 0xd, 0x0)
	/home/tobias/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/split_queue.go:164 +0x65
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*baseQueue).processReplica.func1(0x5492920, 0xc002c13e60, 0xdf8475800, 0x5492920)
	/home/tobias/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/queue.go:964 +0x2ac
github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout(0x5492920, 0xc002c13e60, 0xc00468eb80, 0x1e, 0xdf8475800, 0xc003b36e68, 0x0, 0x0)
	/home/tobias/go/src/github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:140 +0x9e
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*baseQueue).processReplica(0xc004ba03c0, 0x5492960, 0xc004f119b0, 0x55043e0, 0xc0009c3b00, 0x0, 0x0)
	/home/tobias/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/queue.go:923 +0x2b3
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*baseQueue).DrainQueue(0xc004ba03c0, 0xc004f0a780)
	/home/tobias/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/queue.go:1313 +0x12a
github.com/cockroachdb/cockroach/pkg/kv/kvserver.forceScanAndProcess(0xc0045cb500, 0xc004ba03c0, 0xc004f0a780, 0x54928a0)
	/home/tobias/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/queue_helpers_testutil.go:42 +0xd6
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).ForceSplitScanAndProcess(...)
	/home/tobias/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/queue_helpers_testutil.go:73
github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestBackpressureNotAppliedWhenReducingRangeSize.func7.1(0xc0045cb500)
	/home/tobias/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_replica_backpressure_test.go:276 +0x3a
created by github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestBackpressureNotAppliedWhenReducingRangeSize.func7
	/home/tobias/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_replica_backpressure_test.go:276 +0x5ed

--- a/pkg/kv/kvserver/client_replica_backpressure_test.go
+++ b/pkg/kv/kvserver/client_replica_backpressure_test.go
@@ -273,7 +273,10 @@ func TestBackpressureNotAppliedWhenReducingRangeSize(t *testing.T) {
 		require.Len(t, repl.Desc().Replicas().Descriptors(), 1)
 		// We really need to make sure that the split queue has hit this range,
 		// otherwise we'll fail to backpressure.
-		go func() { _ = s.ForceSplitScanAndProcess() }()
+		_ = tc.Stopper().RunAsyncTask(ctx, "force-split", func(context.Context) {
+			_ = s.ForceSplitScanAndProcess()
+		})
+
 		waitForBlocked(repl.RangeID)
 
 		// Observe backpressure now that the range is just over the limit.
@@ -287,11 +290,11 @@ func TestBackpressureNotAppliedWhenReducingRangeSize(t *testing.T) {
 		ctxWithCancel, cancel := context.WithCancel(ctx)
 		defer cancel()
 		upsertErrCh := make(chan error)
-		go func() {
+		_ = tc.Stopper().RunAsyncTask(ctx, "upsert", func(ctx context.Context) {
 			_, err := c.ExecEx(ctxWithCancel, "UPSERT INTO foo VALUES ($1, $2)",
 				nil /* options */, rRand.Intn(numRows), randutil.RandBytes(rRand, rowSize))
 			upsertErrCh <- err
-		}()
+		})
 
 		select {
 		case <-time.After(10 * time.Millisecond):

